src/smolagents/local_python_executor.py

```diff
@@ -394,6 +396,11 @@ def new_func(*args: Any, **kwargs: Any) -> Any:
 
         return result
 
+    # Store original AST, source code, and name
+    new_func.__ast__ = func_def
+    new_func.__source__ = source_code
```

Comment:
These are custom variables, so I would recommend not naming them as dunder (official special attributes), but instead, as private variables:
- `__ast`
- `__source`

Note that I already used `__source` for the `tool` decorator:
- #943